app/src/main/java/org/wikipedia/notifications/NotificationEditorTasksHandler.java

```diff
@@ -0,0 +1,91 @@
+package org.wikipedia.notifications;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.v4.app.NotificationCompat;
+
+import org.wikipedia.Constants;
+import org.wikipedia.R;
+import org.wikipedia.WikipediaApp;
+import org.wikipedia.dataclient.mwapi.EditorTaskCounts;
+import org.wikipedia.editactionfeed.AddTitleDescriptionsActivity;
+import org.wikipedia.events.EditorTaskUnlockEvent;
+import org.wikipedia.settings.Prefs;
+
+import static org.wikipedia.Constants.MIN_LANGUAGES_TO_UNLOCK_TRANSLATION;
+
+public final class NotificationEditorTasksHandler {
+
+    public static void dispatchEditorTaskResults(@NonNull Context context, @NonNull EditorTaskCounts results) {
+        int numTargetsPassed = results.getDescriptionEditTargetsPassed().size();
+        Object eventToDispatch = null;
+
+        if (numTargetsPassed > 1) {
+            if (!Prefs.isEditActionAddDescriptionsUnlocked()) {
```

Comment:
When would this happen?

Reply:
After 50 edits (and after 48 hours of no reverts).

Reply:
Oh, if you meant to say "Why would adding descriptions be locked if the user has already passed 1 target?" the answer is "On a fresh install of the app."

Reply:
I'm confused isn't it 'numTargetsPassed=1 is add descs unlocked' after 5 edits 48hr wait  and then 50 edits 48 hrs 'numTargetsPassed=2 add trans unlocked'?
isn't it, if 2 targets are unlocked then AddDescriptions is definitely unlocked long time back? and even if the user has lost previleges due to reverts... AddDescriptions would first be unlocked before numTargetsPassed > 1 ever happens?

Reply:
> Oh, if you meant to say "Why would adding descriptions be locked if the user has already passed 1 target?" the answer is "On a fresh install of the app."

Ahhh! ding! ding! ding! thank you.. so much logic all organized so neatly with this branch ! merging it now!